src/test/java/cz/muni/fi/pa165/bluebat/entity/TripTest.java

```diff
@@ -0,0 +1,154 @@
+import java.util.List;
+
+/**
+ * @author : Rudolf Madoran
+ * @since : 7. 4. 2021, Wed
+ **/
+
+@Transactional
+@TestExecutionListeners(TransactionalTestExecutionListener.class)
+@ContextConfiguration(classes = PersistenceTravelAgencyApplicationContext.class)
+public class TripTest extends AbstractTestNGSpringContextTests  {
+
+    @PersistenceUnit
+    private EntityManagerFactory emf;
+
+
+
+    @DataProvider(name = "valuesInvalidStrings")
+    private static Object[][] getInvalidStrings() {
+        return new String[][] { { null }, { "" }, { " " }, { "\t" }, { "\n" } };
+    }
+
+    @DataProvider(name = "valuesInvalidDates")
+    private static Object[][] getInvalidDates() {
+
+        return new LocalDate[][] { { null },{LocalDate.of(2020,5,1)} , {LocalDate.of(1878,5,1)} ,
+                {LocalDate.of(1,1,1)}};
+    }
+
+    @DataProvider(name = "valuesInvalidAvailability")
+    private static Object[][] getInvalidAvailability() {
+
+        return new Integer[][] {{null},{-5},{-10}};
+    }
+
+
+    @AfterClass
+    private void removeTrips() {
+        EntityManager em = null;
+        try {
+            em = emf.createEntityManager();
+            em.getTransaction().begin();
+            for (Trip trip: findAll()) {
+                em.remove(em.contains(trip) ? trip : em.merge(trip));
+            }
+            em.getTransaction().commit();
+        } finally {
+            if (em != null) em.close();
+        }
+    }
+
+
+
+    Trip prepareTrip() {
+        Trip testTrip3 = new Trip();
+        testTrip3.setAvailableTrips(5);
+        testTrip3.setDateFrom(LocalDate.of(2020,5,1));
+        testTrip3.setExcursions(new HashSet<Excursion>() {});
+        testTrip3.setDestination("Madrid");
+        testTrip3.setDateTo(LocalDate.of(2020,5,15));
+        return  testTrip3;
+    }
+
+
+    void persistTrip(Trip trip) {
+
+        Trip testTrip3 = new Trip();
+        testTrip3.setAvailableTrips(5);
+        testTrip3.setDateFrom(LocalDate.of(2020,5,1));
+        testTrip3.setExcursions(new HashSet<Excursion>() {});
+        testTrip3.setDestination("Madrid");
+        testTrip3.setDateTo(LocalDate.of(2020,5,15));
+
+        EntityManager em = null;
+        try {
+            em = emf.createEntityManager();
+            em.getTransaction().begin();
+            em.persist(trip);
+            em.getTransaction().commit();
+        } finally {
+            if (em != null) em.close();
+        }
+    }
+    public List<Trip> findAll(){
+        EntityManager em = null;
+        try {
+            em = emf.createEntityManager();
+            em.getTransaction().begin();
+            List<Trip>  list = em.createQuery("select t from Trip t",Trip.class).getResultList() ;
+            em.getTransaction().commit();
+            return list;
+        } finally {
+            if (em != null) em.close();
+        }
+
+    }
+
+    @Test(dataProvider = "valuesInvalidAvailability")
+    public void tripNullDestination(Integer data) {
```

Comment:
Přejmenoval bych názvy testů, aby to odpovídalo realitě, protože se netestuje jen null hodnota. Proto něco jako tripInvalidDestination. Podobně i u ostatních názvů testů.

Navíc v tomto testu testuješ availableTrips, takže by se to mělo jmenovat něco jako tripInvalidAvailableTrips